Remove svg files, the Icon component, and svg-react-loader

afercia opened the pull request:
## Summary

This PR can be summarized in the following changelog entry:

* REmoved the Icon component, replaced by the new SvgIcon React component

## Relevant technical choices:

- this PR is branched out from `stories/415-styled-styled-section`
- removes the Icon component
- removes the svg files from `style-guide/svg`
- uses StyledSection where possible
- adapt other components, e.g. IconButton, IconButtonToggle, to the new SvgIcon component introduced in `stories/415-styled-styled-section`
- removes svg-react-loader

I've looked for all the imports of icons in the other repos and tried to address them. Please help me check no other usages of the SVG files and hte `<Icon>` component have survived and nothing breaks 🙂 
@Xyfi please check the `svg-react-loader` stuff, I have no idea what I've done there 😆 

## Test instructions

- check all the components with icons work correctly in the standalone version
- switch both free and premium to their `stories/415-styled-styled-section` branches if they're not merged yet
- yarn link this branch to free and premium, build the assets, and check all the icons
- on Premium, you will get an error when building the CSS because `StyledSection.scss` was removed, this is covered on free in https://github.com/Yoast/wordpress-seo/pull/8812 I guess your best option is to temporarily remove that import manually

Fixes #420

Xyfi commented on composites/Plugin/Shared/components/HelpCenterButton.js:
```diff
@@ -17,9 +16,9 @@ import { Icon } from "../../Shared/components/Icon";
 	return (
 		<YoastButton { ...props }>
-			<Icon icon={ questionCircle } color={ props.iconColor } />
+			<SvgIcon icon="questionCircle" color={ props.iconColor } />
```
I believe this should be `"question-circle"`.

Xyfi commented on composites/Plugin/Shared/components/SvgIcon.js:
```diff
@@ -0,0 +1,96 @@
+		}
+		/* eslint-enable max-len */
+
+		if ( ! path ) {
```
Maybe add a `console.warn` for easier debugging?

Xyfi commented on composites/Plugin/Shared/components/Icon.js:
```diff
@@ -1,35 +0,0 @@
-import React from "react";
```
I don't think this file has to be removed, because it can still be used to render svg react components. We could still have more complex react components (that use more than just a path) that are just plain react components that are passed to this component.

afercia replied:
OK, however the Icon component had also a IconTest.js, which required the edit.svg file.

afercia replied:
And should we keep `svg-react-loader` then?

Xyfi replied:
No, in case we keep using `Icon.js`, we start using SVG's are react components and not as svg files:

```js
// React component
const EditIcon = ( props ) => {
    return(
        <svg { ...props } >{ "..." }</svg>
    )
}

afercia commented:
Note: I guess this should be rebased interactively after #416 gets merged.

afercia commented:
Instead of building new components for the SVGs, I've adjusted the existing `YoastLogo..js` and `YoastSeoIcon.js` to work with the `<Icon />` component, thus wrapper in a styled components. Building new ones would have led to duplication, as the new components would have been very similar to the existing ones.

Adjusted the usage of the icons in the Onboarding Wizard.

To see hte new icons in action, use this code in `App.js` or yarn-link yoast-components to the plugin ((the MessageBox gets displayed when, for some reasons, the configuration can't be loaded or is empty)


```
import React from "react";
import { IntlProvider } from "react-intl";
import styled from "styled-components";

import Icon from "./composites/Plugin/Shared/components/Icon";
import YoastLogo from "./composites/basic/YoastLogo"
import YoastSeoIcon from "./composites/basic/YoastSeoIcon"

const Container = styled.div`
	width: 800px;
	margin: 0 auto;
	text-align: center;
`;

class App extends React.Component {

	constructor() {
		super();
	}

	render() {
		return (
			<IntlProvider locale="en">
				<Container>
					<Icon icon={ YoastLogo } width="200px" height="93px" />
					<br />
					<Icon icon={ YoastSeoIcon } color="red" width="100px" height="100px" />
				</Container>
			</IntlProvider>
		);
	}
}

export default App;

```

afercia commented:
P.S. I haven't found any other usage of `YoastLogo.js` and `YoastSeoIcon.js` but worth double checking.

afercia commented:
Note: the `color` prop doesn't have any effect when the SVG provides its own fill colors for the shape. It will work for monochromatic icons that don't have a `fill` attribute in the SVG markup.

Xyfi commented:
CR :+1:

igorschoester commented:
I added StyledSection tests base on @afercia 's test `App.js` code in this [PR](https://github.com/Yoast/yoast-components/pull/416). A side effect of this is that a console warning is shown on the test with a non-existing icon 😅 

Acceptance 👍
